x-pack/platform/plugins/shared/fleet/public/applications/fleet/sections/settings/components/fleet_server_hosts_flyout/index.tsx

```diff
@@ -230,6 +231,127 @@ export const FleetServerHostsFlyout: React.FunctionComponent<FleetServerHostsFly
               }
             />
           </EuiFormRow>
+          <EuiSpacer size="l" />
+          <EuiFormRow
+            fullWidth
+            {...inputs.certificateInput.formRowProps}
+            label={
+              <FormattedMessage
+                id="xpack.fleet.settings.fleetServerHostsFlyout.certificateLabel"
+                defaultMessage="Client SSL Certificate"
+              />
+            }
+          >
+            <EuiFieldText
+              fullWidth
+              data-test-subj="fleetServerHostsFlyout.certificateInput"
+              placeholder={i18n.translate(
+                'xpack.fleet.settings.fleetServerHostsFlyout.certificatePlaceholder',
+                { defaultMessage: 'Specify ssl certificate' }
+              )}
+              {...inputs.certificateInput.props}
+            />
+          </EuiFormRow>
+          <EuiFormRow
+            fullWidth
+            {...inputs.certificateKeyInput.formRowProps}
+            label={
+              <FormattedMessage
+                id="xpack.fleet.settings.fleetServerHostsFlyout.certificateKeyLabel"
+                defaultMessage="Client SSL Certificate key"
+              />
+            }
+          >
+            <EuiFieldText
+              fullWidth
+              data-test-subj="fleetServerHostsFlyout.certificateKeyInput"
+              placeholder={i18n.translate(
+                'xpack.fleet.settings.fleetServerHostsFlyout.certificateKeyPlaceholder',
+                { defaultMessage: 'Specify SSL certificate key' }
+              )}
+              {...inputs.certificateKeyInput.props}
+            />
+          </EuiFormRow>
+          <EuiFormRow
+            fullWidth
+            {...inputs.certificateAuthoritiesInput.formRowProps}
+            label={
+              <FormattedMessage
+                id="xpack.fleet.settings.fleetServerHostsFlyout.certificateAuthoritiesLabel"
+                defaultMessage="Server SSL certificate authorities (optional)"
```

Comment:
Aren't all fields optional? I'm wondering why some of them have `(optional)` label

Reply:
Yes they are, this was in the original requirements from @nimarezainia. I'll remove it for now and I'll add it back if needed.

Reply:
the ones marked optional are only needed if the user required mTLS. but you are right that here they are all optional.